gflownet/envs/base.py

```diff
@@ -223,14 +258,23 @@ def _pre_step(
             raise ValueError(
                 f"Tried to execute action {action} not present in action space."
             )
-        # If env is done, step should not proceed.
-        if self.done:
-            return False, self.state, action
+        # If backward and state is source, step should not proceed.
+        if backward is True:
```

Comment:
```suggestion
        if backward:
```
Or is there a specific reason to have `is`?

Reply:
I have two reasons: 

- to avoid confusion with truthy or falsy values.
- to make it explicit what is being checked.

Is that bad or not pythonic?

Reply:
I don't think it's a problem, it's a little non-pythonic I'd say (and to me it would normally indicate that there is a real possibility that values can be something other than boolean, and that has to be treated separately). I was wondering if that's actually one of these cases.

Reply:
Ok, happy to change to what is more pythonic or more common, but there are many many of such cases, so maybe let's change later as this is not urgent.

Reply:
I think this is the realm of subjective opinion. I would also go for `is backward` and enforce type checking on backward somewhere else to ensure it is a boolean always, but you could make an argument for being explicit everywhere as you have done.